code/game/objects/items/weapons/cards_ids.dm

```diff
@@ -354,262 +363,265 @@
+		return
+	age = new_age
+
+/obj/item/card/id/syndicate/proc/change_occupation()
+	var/list/departments =list(
+				"Assistant",
+				"Engineering",
+				"Medical",
+				"Science",
+				"Security",
+				"Service",
+				"Command",
+				"Special",
+				"Custom",
+			)
+
+	var/new_job
+	var/new_rank
+
+	var/tgui_message = "What job would you like to put on this card?"
+	var/department = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", departments)
+	if(department != "Custom")
+		switch(department)
+			if("Engineering")
+				new_job = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.engineering_positions)
+			if("Medical")
+				new_job = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.medical_positions)
+			if("Science")
+				new_job = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.science_positions)
+			if("Security")
+				new_job = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.security_positions)
+			if("Service")
+				new_job = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.service_positions)
+			if("Command")
+				new_job = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.command_positions)
+			if("Special")
+				new_job = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", (get_all_solgov_jobs() + get_all_soviet_jobs() + get_all_centcom_jobs()))
+			if("Assistant")
+				new_job = "Assistant"
+		new_rank = new_job
 	else
-		..()
+		new_job = sanitize(tgui_input_text(registered_human,"Choose a custom job title:", "Agent Card Occupation", "Assistant", MAX_MESSAGE_LEN))
+		tgui_message = "What SecHUD icon would you like to be shown on this card?"
+		var/department_icon = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", departments)
+		switch(department_icon)
+			if("Engineering")
+				new_rank = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.engineering_positions)
+			if("Medical")
+				new_rank = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.medical_positions)
+			if("Science")
+				new_rank = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.science_positions)
+			if("Security")
+				new_rank = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.security_positions)
+			if("Service")
+				new_rank = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.service_positions)
+			if("Command")
+				new_rank = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", GLOB.command_positions)
+			if("Special")
+				new_rank = tgui_input_list(registered_human, tgui_message, "Agent Card Occupation", (get_all_solgov_jobs() + get_all_soviet_jobs() + get_all_centcom_jobs()))
+			if("Custom")
+				new_rank = null
+			if("Assistant")
+				new_rank = "Assistant"
```

Comment:
I know this is how the old agent ID worked, but imo it should really be entirely redone. Ideally how would work is instead of there being these repeated long lists, there is two buttons, the old button will pop up a static list of all jobs (instead of doing the department based approach). And a second button (maybe bigger than I pictured here for setting custom job/picture. Seperating these out into seperate buttons would make it a lot easier to interact with.
![image](https://github.com/ParadiseSS13/Paradise/assets/91113370/a5f22dc2-4d57-4fbc-ad86-fd440ab11628)

The static list for the normal jobs can be created in Initialize with
```
var/static/list/possible_jobs
...
if(!possible_jobs)
        possible_jobs = sortTim(GLOB.joblist + get_all_solgov_jobs() + get_all_soviet_jobs() + get_all_centcom_jobs()), GLOBAL_PROC_REF(cmp_text_asc))
```
Additionally, when you do a custom job, it should just give you a text prompt for the job name, and then a second prompt to choose from all possible job hud icons. You can also create a static list of these by using `GLOB.joblist + "Prisoner" + "Centcom" + "Solgov" + "Soviet"`